lib/auth/Vault.ts

```diff
@@ -356,6 +358,35 @@ export default class Vault {
             });
     }
 
+    /**
+     * A getter for account canonical IDs given a list of account IDs
+     * @param accountIds - list of account IDs
+     * @param log - log object
+     * @param callback - callback function
+     * @returns callback with either error or an object from Vault
+     * containing canonicalID and display name for each account ID
+     */
+    getCanonicalIdsByAccountIds(
+        accountIds: string[],
+        log: RequestLogger,
+        callback: GenericCallback<AccountCannonicalInfo[]>,
+    ) {
+        log.addDefaultFields({ accountIds, method: 'getCanonicalIdsByAccountIds' });
```

Comment:
I am not sure we want to add a default field on the `method`, as the logger object will be shared with the S3 API call, that might also use this method field. If the field becomes default, then it might make debugging harder.

If the list of account id is large, this might also have an impact, why not only logging this in case of an error?